infinispan-remote/src/test/java/org/hibernate/ogm/datastore/infinispanremote/utils/InfinispanRemoteTestHelper.java

```diff
@@ -175,6 +200,28 @@ public long getNumberOfEntities(SessionFactory sessionFactory) {
 		return counter.get();
 	}
 
+	@Override
+	public long getNumberOfEntities(Session session) {
+		final InfinispanRemoteDatastoreProvider datastoreProvider = getProvider( session );
+		final SessionFactoryImplementor sessionFactoryImplementor = getSessionFactoryImplementor( session.getSessionFactory() );
+		final Collection<EntityPersister> persisters = sessionFactoryImplementor.getMetamodel().entityPersisters().values();
+
+		final AtomicLong counter = new AtomicLong();
+		for ( EntityPersister ep : persisters ) {
+			OgmEntityPersister persister = (OgmEntityPersister) ep;
+			String tableName = persister.getTableName();
+			RemoteCache<Object, Object> cache = datastoreProvider.getCache( tableName );
+			if ( !( cache instanceof TransactionalRemoteCacheImpl ) ) {
+				counter.addAndGet( cache.size() );
+				continue;
+			}
+
+			int size = TransactionalRemoteCacheCounter.count( persister, (TransactionalRemoteCacheImpl) cache, datastoreProvider.getDataMapperForCache( tableName ), session );
+			counter.addAndGet( size );
```

Comment:
maybe a bit of explanation here might be helpful for the future. Like, why the `if`?

Reply:
Wouldn't be better to have a separate method to get the size and then add the value to the counter?